src/15-led-compass/src/main.rs

```diff
@@ -3,14 +3,32 @@
 #![no_std]
 
 #[allow(unused_imports)]
-use aux15::{entry, iprint, iprintln, prelude::*};
+use aux15::{entry, iprint, iprintln, prelude::*, switch_hal::OutputSwitch, Direction, I16x3};
 
 #[entry]
 fn main() -> ! {
-    let (_leds, mut lsm303dlhc, mut delay, mut itm) = aux15::init();
+    let (leds, mut lsm303dlhc, mut delay, _itm) = aux15::init();
+    let mut leds = leds.into_array();
 
     loop {
-        iprintln!(&mut itm.stim[0], "{:?}", lsm303dlhc.mag().unwrap());
+        let I16x3 { x, y, .. } = lsm303dlhc.mag().unwrap();
```

Comment:
From a learning point of view, I think it makes more sense to keep the first state of the program here instead of the solution. What do you think?